group-actors/src/main/java/org/sunbird/actors/SearchGroupActor.java

```diff
@@ -40,14 +49,55 @@ public void onReceive(Request request) throws Throwable {
    */
   private void searchGroup(Request request) throws BaseException {
     logger.info("SearchGroup method call");
+    boolean isUseridRedisEnabled = Boolean.parseBoolean(PropertiesCache.getInstance().getConfigValue(JsonKey.ENABLE_USERID_REDIS_CACHE));
+    CacheUtil cacheUtil = new CacheUtil();
     GroupService groupService = new GroupServiceImpl();
     Map<String, Object> searchQueryMap = request.getRequest();
     Map<String, Object> filterMap = (Map<String, Object>) searchQueryMap.get(JsonKey.FILTERS);
-    List<GroupResponse> groupDetails = groupService.searchGroup(filterMap);
+
+    List<GroupResponse> groupDetails = new ArrayList<>();
+    String userId = (String) filterMap.get(JsonKey.USER_ID);
+    if (StringUtils.isNotBlank(userId)) {
+      boolean getFromDB = false;
+      if(isUseridRedisEnabled) {
+        String groupList = cacheUtil.getCache(userId);
+        if (StringUtils.isNotEmpty(groupList)) {
+          try {
+            groupDetails = JsonUtils.deserialize(groupList, new TypeReference<List<GroupResponse>>() {});
+          } catch (Exception e) {
+            logger.error("Error in getting group list from Redis: {}", e.getMessage());
+            getFromDB = true;
```

Comment:
This pattern of else with true - could be defaulted. And lessen the if-depth.

Reply:
Corrected.